Misleading English message for error TS1046

<!-- BUGS: Please use this template. -->
<!-- QUESTIONS: This is not a general support forum! Ask Qs at http://stackoverflow.com/questions/tagged/typescript -->
<!-- SUGGESTIONS: See https://github.com/Microsoft/TypeScript-wiki/blob/master/Writing-Good-Design-Proposals.md -->

<!-- Please try to reproduce the issue with `typescript@next`. It may have already been fixed. -->
**TypeScript Version:**  2.0.10 (checked; message is the same on master branch)

**Code**

The following test.d.ts file:

    module test {
        export const test: number;
    }

when compiled, yields the following error message:

```error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.```

This is both misleading and just wrong, as the following test.d.ts file is legal:

    export const test: number;

**Expected behavior:**

The error message should not be misleading.

**Actual behavior:**

The error message is misleading. This led me on a wild goose chase trying to get my .d.ts file to compile, all the time using the apparently required `declare` statement, when the solution didn't have `declare` at all.

**Related:**

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -31841,7 +31841,7 @@ namespace ts {
                     return false;
             }
 
-            return grammarErrorOnFirstToken(node, Diagnostics.A_declare_modifier_is_required_for_a_top_level_declaration_in_a_d_ts_file);
+            return grammarErrorOnFirstToken(node, Diagnostics.Top_level_declarations_in_d_ts_files_must_start_with_either_a_declare_or_export_modifier);
         }
 
         function checkGrammarTopLevelElementsForRequiredDeclareModifier(file: SourceFile): boolean {

--- a/src/compiler/diagnosticMessages.json
+++ b/src/compiler/diagnosticMessages.json
@@ -139,7 +139,7 @@
         "category": "Error",
         "code": 1045
     },
-    "A 'declare' modifier is required for a top level declaration in a .d.ts file.": {
+    "Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.": {
         "category": "Error",
         "code": 1046
     },

--- a/tests/baselines/reference/missingRequiredDeclare.d.errors.txt
+++ b/tests/baselines/reference/missingRequiredDeclare.d.errors.txt
@@ -1,10 +1,10 @@
-tests/cases/compiler/missingRequiredDeclare.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/compiler/missingRequiredDeclare.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 tests/cases/compiler/missingRequiredDeclare.d.ts(1,9): error TS1039: Initializers are not allowed in ambient contexts.
 
 
 ==== tests/cases/compiler/missingRequiredDeclare.d.ts (2 errors) ====
     var x = 1;
     ~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
             ~
 !!! error TS1039: Initializers are not allowed in ambient contexts.

--- a/tests/baselines/reference/parserEnumDeclaration3.d.errors.txt
+++ b/tests/baselines/reference/parserEnumDeclaration3.d.errors.txt
@@ -1,9 +1,9 @@
-tests/cases/conformance/parser/ecmascript5/EnumDeclarations/parserEnumDeclaration3.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/EnumDeclarations/parserEnumDeclaration3.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/EnumDeclarations/parserEnumDeclaration3.d.ts (1 errors) ====
     enum E {
     ~~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
        A = 1
     }

--- a/tests/baselines/reference/parserFunctionDeclaration2.d.errors.txt
+++ b/tests/baselines/reference/parserFunctionDeclaration2.d.errors.txt
@@ -1,11 +1,11 @@
-tests/cases/conformance/parser/ecmascript5/FunctionDeclarations/parserFunctionDeclaration2.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/FunctionDeclarations/parserFunctionDeclaration2.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 tests/cases/conformance/parser/ecmascript5/FunctionDeclarations/parserFunctionDeclaration2.d.ts(1,14): error TS1183: An implementation cannot be declared in ambient contexts.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/FunctionDeclarations/parserFunctionDeclaration2.d.ts (2 errors) ====
     function F() {
     ~~~~~~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
                  ~
 !!! error TS1183: An implementation cannot be declared in ambient contexts.
     }

--- a/tests/baselines/reference/parserModuleDeclaration1.d.errors.txt
+++ b/tests/baselines/reference/parserModuleDeclaration1.d.errors.txt
@@ -1,8 +1,8 @@
-tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration1.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration1.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration1.d.ts (1 errors) ====
     module "Foo" {
     ~~~~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
     }

--- a/tests/baselines/reference/parserModuleDeclaration2.d.errors.txt
+++ b/tests/baselines/reference/parserModuleDeclaration2.d.errors.txt
@@ -1,8 +1,8 @@
-tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration2.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration2.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration2.d.ts (1 errors) ====
     module M {
     ~~~~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
     }

--- a/tests/baselines/reference/parserModuleDeclaration4.d.errors.txt
+++ b/tests/baselines/reference/parserModuleDeclaration4.d.errors.txt
@@ -1,11 +1,11 @@
-tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration4.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration4.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration4.d.ts(2,3): error TS1038: A 'declare' modifier cannot be used in an already ambient context.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/ModuleDeclarations/parserModuleDeclaration4.d.ts (2 errors) ====
     module M {
     ~~~~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
       declare module M1 {
       ~~~~~~~
 !!! error TS1038: A 'declare' modifier cannot be used in an already ambient context.

--- a/tests/baselines/reference/parserVariableStatement1.d.errors.txt
+++ b/tests/baselines/reference/parserVariableStatement1.d.errors.txt
@@ -1,10 +1,10 @@
-tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement1.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement1.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement1.d.ts(1,9): error TS1039: Initializers are not allowed in ambient contexts.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement1.d.ts (2 errors) ====
     var v = 1;
     ~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
             ~
 !!! error TS1039: Initializers are not allowed in ambient contexts.

--- a/tests/baselines/reference/parserVariableStatement2.d.errors.txt
+++ b/tests/baselines/reference/parserVariableStatement2.d.errors.txt
@@ -1,7 +1,7 @@
-tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement2.d.ts(1,1): error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement2.d.ts(1,1): error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.
 
 
 ==== tests/cases/conformance/parser/ecmascript5/Statements/parserVariableStatement2.d.ts (1 errors) ====
     var v;
     ~~~
-!!! error TS1046: A 'declare' modifier is required for a top level declaration in a .d.ts file.
+!!! error TS1046: Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.

